src/sentry/explore/translation/alerts_translation.py

```diff
@@ -0,0 +1,211 @@
+import logging
+
+from django.db import router, transaction
+
+from sentry import features
+from sentry.discover.translation.mep_to_eap import QueryParts, translate_mep_to_eap
+from sentry.incidents.models.alert_rule import AlertRuleDetectionType
+from sentry.incidents.subscription_processor import MetricIssueDetectorConfig
+from sentry.incidents.utils.types import DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION
+from sentry.seer.anomaly_detection.store_data import SeerMethod
+from sentry.seer.anomaly_detection.store_data_workflow_engine import (
+    handle_send_historical_data_to_seer,
+)
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.models import (
+    ExtrapolationMode,
+    QuerySubscription,
+    SnubaQuery,
+    SnubaQueryEventType,
+)
+from sentry.snuba.tasks import update_subscription_in_snuba
+from sentry.utils.db import atomic_transaction
+from sentry.workflow_engine.models.data_condition import DataCondition
+from sentry.workflow_engine.models.data_source import DataSource
+
+logger = logging.getLogger(__name__)
+
+
+def snapshot_snuba_query(snuba_query: SnubaQuery):
+    if snuba_query.dataset in [Dataset.PerformanceMetrics.value, Dataset.Transactions.value]:
+        query_snapshot = {
+            "type": snuba_query.type,
+            "dataset": snuba_query.dataset,
+            "query": snuba_query.query,
+            "aggregate": snuba_query.aggregate,
+            "time_window": snuba_query.time_window,
+        }
+        snuba_query.query_snapshot = query_snapshot
+        snuba_query.save()
+
+    return snuba_query
+
+
+def _get_old_query_info(snuba_query: SnubaQuery):
+    old_query_type = SnubaQuery.Type(snuba_query.type)
+    old_dataset = Dataset(snuba_query.dataset)
+    old_query = snuba_query.query
+    old_aggregate = snuba_query.aggregate
+
+    return old_query_type, old_dataset, old_query, old_aggregate
+
+
+def translate_detector_and_update_subscription_in_snuba(snuba_query: SnubaQuery):
+    data_source: DataSource = DataSource.objects.get(
+        source_id=snuba_query.id, type=DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION
```

Comment:
I think django might do some magic to deal with this but technically `DataSource.source_id` is a text field if you want to convert `snuba_query.id` to one